---
id: command-line-interface
sidebar_label: Command Line Interface
title: Command Line Interface
---

## Cheat Sheet

The command line interface (CLI) gives you easy-to-remember commands for common tasks.

|        Command         |                                                                  Effect                                                                  |
|------------------------|------------------------------------------------------------------------------------------------------------------------------------------|
|`rasa init`             |Creates a new project with example training data, actions, and config files.                                                              |
|`rasa train`            |Trains a model using your NLU data and stories, saves trained model in `./models`.                                                        |
|`rasa interactive`      |Starts an interactive learning session to create new training data by chatting.                                                           |
|`rasa shell`            |Loads your trained model and lets you talk to your assistant on the command line.                                                         |
|`rasa run`              |Starts a Rasa server with your trained model. See the [Configuring the HTTP API](/docs/user-guide/configuring-http-api) docs for details. |
|`rasa run actions`      |Starts an action server using the Rasa SDK.                                                                                               |
|`rasa visualize`        |Visualizes stories.                                                                                                                       |
|`rasa test`             |Tests a trained Rasa model using your test NLU data and stories.                                                                          |
|`rasa data split nlu`   |Performs a split of your NLU data according to the specified percentages.                                                                 |
|`rasa data convert nlu` |Converts NLU training data between different formats.                                                                                     |
|`rasa export`           |Export conversations from a tracker store to an event broker.                                                                             |
|`rasa x`                |Launch Rasa X locally.                                                                                                                    |
|`rasa -h`               |Shows all available commands.                                                                                                             |

## Create a new project

A single command sets up a complete project for you with some example training data.

```bash
rasa init
```

This creates the following files:

```bash
.
├── __init__.py
├── actions.py
├── config.yml
├── credentials.yml
├── data
│   ├── nlu.md
│   └── stories.md
├── domain.yml
├── endpoints.yml
├── models
│   └── <timestamp>.tar.gz
└── tests
   └── conversation_tests.md
```

The `rasa init` command will ask you if you want to train an initial model using this data.
If you answer no, the `models` directory will be empty.

With this project setup, common commands are very easy to remember.
To train a model, type `rasa train`, to talk to your model on the command line, `rasa shell`,
to test your model type `rasa test`.

## Train a Model

The main command is:

```bash
rasa train
```

This command trains a Rasa model that combines a Rasa NLU and a Rasa Core model.
If you only want to train an NLU or a Core model, you can run `rasa train nlu` or `rasa train core`.
However, Rasa will automatically skip training Core or NLU if the training data and config haven’t changed.

`rasa train` will store the trained model in the directory defined by `--out`. The name of the model
is per default `<timestamp>.tar.gz`. If you want to name your model differently, you can specify the name
using `--fixed-model-name`.

The following arguments can be used to configure the training process:

```text
Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (ujson 3.0.0 (/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages), Requirement.parse('ujson<3.0,>=1.35'), {'rasa'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/bin/rasa", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3258, in <module>
    @_call_aside
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3242, in _call_aside
    f(*args, **kwargs)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3271, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 586, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 599, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 787, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'ujson<3.0,>=1.35' distribution was not found and is required by rasa
```

:::note
Make sure training data for Core and NLU are present when training a model using `rasa train`.
If training data for only one model type is present, the command automatically falls back to
`rasa train nlu` or `rasa train core` depending on the provided training files.

:::

## Interactive Learning

To start an interactive learning session with your assistant, run

```bash
rasa interactive
```

If you provide a trained model using the `--model` argument, the interactive learning process
is started with the provided model. If no model is specified, `rasa interactive` will
train a new Rasa model with the data located in `data/` if no other directory was passed to the
`--data` flag. After training the initial model, the interactive learning session starts.
Training will be skipped if the training data and config haven’t changed.

The full list of arguments that can be set for `rasa interactive` is:

```text
Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (ujson 3.0.0 (/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages), Requirement.parse('ujson<3.0,>=1.35'), {'rasa'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/bin/rasa", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3258, in <module>
    @_call_aside
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3242, in _call_aside
    f(*args, **kwargs)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3271, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 586, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 599, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 787, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'ujson<3.0,>=1.35' distribution was not found and is required by rasa
```

## Talk to your Assistant

To start a chat session with your assistant on the command line, run:

```bash
rasa shell
```

The model that should be used to interact with your bot can be specified by `--model`.
If you start the shell with an NLU-only model, `rasa shell` allows
you to obtain the intent and entities of any text you type on the command line.
If your model includes a trained Core model, you can chat with your bot and see
what the bot predicts as a next action.
If you have trained a combined Rasa model but nevertheless want to see what your model
extracts as intents and entities from text, you can use the command `rasa shell nlu`.

To increase the logging level for debugging, run:

```bash
rasa shell --debug
```

:::note
In order to see the typical greetings and/or session start behavior you might see
in an external channel, you will need to explicitly send `/session_start`
as the first message. Otherwise, the session start behavior will begin as described in
[Session configuration](/docs/core/domains#session-config).

:::

The full list of options for `rasa shell` is:

```text
Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (ujson 3.0.0 (/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages), Requirement.parse('ujson<3.0,>=1.35'), {'rasa'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/bin/rasa", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3258, in <module>
    @_call_aside
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3242, in _call_aside
    f(*args, **kwargs)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3271, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 586, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 599, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 787, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'ujson<3.0,>=1.35' distribution was not found and is required by rasa
```

## Start a Server

To start a server running your Rasa model, run:

```bash
rasa run
```

The following arguments can be used to configure your Rasa server:

```text
Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (ujson 3.0.0 (/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages), Requirement.parse('ujson<3.0,>=1.35'), {'rasa'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/bin/rasa", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3258, in <module>
    @_call_aside
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3242, in _call_aside
    f(*args, **kwargs)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3271, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 586, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 599, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 787, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'ujson<3.0,>=1.35' distribution was not found and is required by rasa
```

For more information on the additional parameters, see [Configuring the HTTP API](/docs/user-guide/configuring-http-api).
See the Rasa [HTTP API](/docs/api/http-api) docs for detailed documentation of all the endpoints.

<a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor" id="run-action-server"></a>

## Start an Action Server

To run your action server run

```bash
rasa run actions
```

The following arguments can be used to adapt the server settings:

```text
Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (ujson 3.0.0 (/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages), Requirement.parse('ujson<3.0,>=1.35'), {'rasa'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/bin/rasa", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3258, in <module>
    @_call_aside
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3242, in _call_aside
    f(*args, **kwargs)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3271, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 586, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 599, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 787, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'ujson<3.0,>=1.35' distribution was not found and is required by rasa
```

## Visualize your Stories

To open a browser tab with a graph showing your stories:

```bash
rasa visualize
```

Normally, training stories in the directory `data` are visualized. If your stories are located
somewhere else, you can specify their location with `--stories`.

Additional arguments are:

```text
Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (ujson 3.0.0 (/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages), Requirement.parse('ujson<3.0,>=1.35'), {'rasa'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/bin/rasa", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3258, in <module>
    @_call_aside
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3242, in _call_aside
    f(*args, **kwargs)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3271, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 586, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 599, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 787, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'ujson<3.0,>=1.35' distribution was not found and is required by rasa
```

## Evaluating a Model on Test Data

To evaluate your model on test data, run:

```bash
rasa test
```

Specify the model to test using `--model`.
Check out more details in [Evaluating an NLU Model](/docs/user-guide/testing-your-assistant#nlu-evaluation) and [Evaluating a Core Model](/docs/user-guide/testing-your-assistant#core-evaluation).

The following arguments are available for `rasa test`:

```text
Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (ujson 3.0.0 (/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages), Requirement.parse('ujson<3.0,>=1.35'), {'rasa'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/bin/rasa", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3258, in <module>
    @_call_aside
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3242, in _call_aside
    f(*args, **kwargs)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3271, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 586, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 599, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 787, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'ujson<3.0,>=1.35' distribution was not found and is required by rasa
```

<a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor" id="train-test-split"></a>

## Create a Train-Test Split

To create a split of your NLU data, run:

```bash
rasa data split nlu
```

You can specify the training data, the fraction, and the output directory using the following arguments:

```text
Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (ujson 3.0.0 (/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages), Requirement.parse('ujson<3.0,>=1.35'), {'rasa'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/bin/rasa", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3258, in <module>
    @_call_aside
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3242, in _call_aside
    f(*args, **kwargs)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3271, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 586, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 599, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 787, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'ujson<3.0,>=1.35' distribution was not found and is required by rasa
```

This command will attempt to keep the proportions of intents the same in train and test.
If you have NLG data for retrieval actions, this will be saved to seperate files:

```bash
ls train_test_split

      nlg_test_data.md     test_data.json
      nlg_training_data.md training_data.json
```

## Convert Data Between Markdown and JSON

To convert NLU data from LUIS data format, WIT data format, Dialogflow data format, JSON, or Markdown
to JSON or Markdown, run:

```bash
rasa data convert nlu
```

You can specify the input file, output file, and the output format with the following arguments:

```text
Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (ujson 3.0.0 (/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages), Requirement.parse('ujson<3.0,>=1.35'), {'rasa'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/bin/rasa", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3258, in <module>
    @_call_aside
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3242, in _call_aside
    f(*args, **kwargs)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3271, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 586, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 599, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 787, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'ujson<3.0,>=1.35' distribution was not found and is required by rasa
```

<a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor" id="section-export"></a>

## Export Conversations to an Event Broker

To export events from a tracker store using an event broker, run:

```bash
rasa export
```

You can specify the location of the environments file, the minimum and maximum
timestamps of events that should be published, as well as the conversation IDs that
should be published.

```text
Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (ujson 3.0.0 (/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages), Requirement.parse('ujson<3.0,>=1.35'), {'rasa'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/bin/rasa", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3258, in <module>
    @_call_aside
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3242, in _call_aside
    f(*args, **kwargs)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3271, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 586, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 599, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 787, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'ujson<3.0,>=1.35' distribution was not found and is required by rasa
```

<a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor" id="section-evaluation"></a>

## Start Rasa X

Rasa X is a toolset that helps you leverage conversations to improve your assistant.
You can find more information about it <a class="reference external" href="https://rasa.com/docs/rasa-x/" target="_blank">here</a>.You can start Rasa X locally by executing

```bash
rasa x
```

To be able to start Rasa X you need to have Rasa X local mode installed
and you need to be in a Rasa project.:::note
By default Rasa X runs on the port 5002. Using the argument `--rasa-x-port` allows you to change it to
any other port.

:::

The following arguments are available for `rasa x`:

```text
Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (ujson 3.0.0 (/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages), Requirement.parse('ujson<3.0,>=1.35'), {'rasa'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/ella/miniconda3/envs/rasa/bin/rasa", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3258, in <module>
    @_call_aside
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3242, in _call_aside
    f(*args, **kwargs)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3271, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 586, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 599, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/Users/ella/miniconda3/envs/rasa/lib/python3.7/site-packages/pkg_resources/__init__.py", line 787, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'ujson<3.0,>=1.35' distribution was not found and is required by rasa
```
